compositor_pipeline/src/queue.rs

```diff
@@ -69,6 +67,20 @@ pub struct ScheduledEvent {
     callback: Box<dyn FnOnce() + Send>,
 }
 
+pub enum PipelineEvent<T> {
+    Data(T),
+    EOS,
+}
```

Comment:
Shouldn't this be defined in `pipeline`?

Reply:
It could be, I don't have strong opinion. I placed it here to contrast with ScheduledEvent above